src/devices/bus/sega8/sega8_slot.h

```diff
@@ -330,7 +335,12 @@ class sms_card_slot_device : public sega8_card_slot_device
 	sms_card_slot_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
 	virtual bool must_be_loaded() const noexcept override { return false; }
 	virtual const char *image_interface() const noexcept override { return "sms_card"; }
-	virtual const char *file_extensions() const noexcept override { return "bin"; }
+	virtual const char *file_extensions() const noexcept override { return "bin,sg"; }
+
+	// required for the 3D glasses
+	DECLARE_WRITE_LINE_MEMBER(write_sscope) { if (m_cart) m_cart->write_sscope(state); }
+	template <typename T> void set_screen_tag(T &&tag) { m_screen.set_tag(std::forward<T>(tag)); }
+	required_device<screen_device> m_screen;
```

Comment:
Please use access levels properly, clearly separate configuration and live input member functions, and don’t mix data members with member functions.

Reply:
The changes I sent today reorganizes the code to make it more restrictive and tries to follow your guidance.